src/VestRewardsDistributionJob.sol

```diff
@@ -0,0 +1,150 @@
+    error RewardDistributionDoesNotExist(address farm);
+
+    // --- Events ---
+    event Work(bytes32 indexed network, address[] rewDist, uint[] distAmounts);
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event AddRewardDistribution(address indexed rewdist);
+    event RemoveRewardDistribution(address indexed rewDist);
+
+    constructor(
+        address _sequencer
+    ) {
+        sequencer = SequencerLike(_sequencer);
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    // --- Reward Distribution Admin ---
+    function addRewardDistribution(address rewdist) external auth {
+        if (!distributions.add(rewdist)) revert RewardDistributionExists(rewdist);
+        emit AddRewardDistribution(rewdist);
+    }
+
+    function removeRewardDistribution(address rewdist) external auth {
+        if (!distributions.remove(rewdist)) revert RewardDistributionDoesNotExist(rewdist);
+        emit RemoveRewardDistribution(rewdist);
+    }
+
+    function work(bytes32 network, bytes calldata args) public {
+        if (!sequencer.isMaster(network)) revert NotMaster(network);
+
+        (address[] memory rewDistributions) = abi.decode(args, (address[]));
+
+        if (rewDistributions.length > 0){
+            uint256[] memory distAmounts = new uint256[](rewDistributions.length);
+            for (uint256 i = 0; i < rewDistributions.length; i++) {
+                address rewDist = rewDistributions[i];
+                // prevent keeper from calling random contracts having distribute()
+                if (!distributions.contains(rewDist)) revert RewardDistributionDoesNotExist(rewDist);
+                distAmounts[i] = VestedRewardsDistributionLike(rewDistributions[i]).distribute();
+            }
+            emit Work(network, rewDistributions, distAmounts);
+        }
+        else {
+            revert NothingToDistribute();
+        }
+    }
+
+    function workable(bytes32 network) external view override returns (bool, bytes memory) {
+        if (!sequencer.isMaster(network)) return (false, bytes("Network is not master"));
+
+        uint256 distributionsLen = distributions.length();
+        if (distributionsLen > 0) {
+            address[] memory distributable = new address[](distributionsLen);
+            uint256 numFarms;
+            for (uint256 i = 0; i < distributionsLen; i++) {
+                address rewDist = distributions.at(i);
+                uint256 vestId = VestedRewardsDistributionLike(rewDist).vestId();
+                DssVestWithGemLike dssVest = VestedRewardsDistributionLike(rewDist).dssVest();
+                uint256 amount = dssVest.unpaid(vestId);
+                if (amount > 0) {
+                    ++numFarms;
+                    distributable[i] = rewDist;
+                }
+            }
+            address[] memory rewDistributions = new address[](numFarms);
+            uint256 j;
+            for (uint256 i = 0; i < distributionsLen; i++) {
```

Comment:
Shouldn't this iteration go until `numFarms` instead?

Reply:
This is what i had done in the beginning, the bug is: let's  say we have 2 farms with unpaid amounts [0,100] then if the loop up to `numFarms` we would miss the 2nd because there would be one iteration. The tests caught that bug. Having said that, there can be a more efficient implementation based on you comment above